store/src/java/com/zimbra/cs/extension/ExtensionDispatcherServlet.java

```diff
@@ -129,19 +136,17 @@ public void service(HttpServletRequest req, HttpServletResponse resp) throws IOE
      */
     public static void unregister(ZimbraExtension ext) {
         synchronized(sHandlers) {
-            for (Iterator it = sHandlers.keySet().iterator(); it.hasNext(); ) {
-                String path = (String) it.next();
-                if (path.startsWith("/" + ext.getName())) {
-                    ExtensionHttpHandler handler = (ExtensionHttpHandler) sHandlers.get(path);
-                    try {
-                        handler.destroy();
-                    } catch (Exception e) {
-                        ZimbraLog.extensions.warn("Error in destroy for handler " + handler.getClass(), e);
-                    } finally {
-                        it.remove();
-                    }
+            String extName = ext.getName();
+            for (ExtensionHttpHandler extHandler: EXTENSION_HANDLERS.get(extName)) {
+                try {
+                    extHandler.destroy();
+                } catch (Exception e) {
+                    ZimbraLog.extensions.warn("Error in destroy for handler " + extHandler.getClass(), e);
+                } finally {
```

Comment:
Use `try-with-resources `here, as it automatically handles cleanup, improves code readability, enhances exception handling, reduces resource leak risks, and simplifies null checking compared to traditional `finally` blocks.

Reply:
That's good in other cases. but here it seems `try-catch-finally` is the best choice. since we are not opening try-on `sHandlers`